front_end/models/react_native/ReactDevToolsBindingsModel.ts

```diff
@@ -54,10 +56,44 @@ export class ReactDevToolsBindingsModel extends SDK.SDKModel.SDKModel {
     }
 
     if (parsedMessage) {
-      this.dispatchMessageToDomainEventListeners(parsedMessage.domain, parsedMessage.message);
+      const domainName = parsedMessage.domain;
+
+      // Messages are flushed out when execution context is created and dispatcher is initialized
+      if (this.fuseboxDispatcherIsInitialized && this.isDomainMessagesQueueEmpty(domainName)) {
+        this.dispatchMessageToDomainEventListeners(domainName, parsedMessage.message);
+      } else {
+        // This could happen when backend is already sending messages via binding
+        // But ReactDevToolsBindingsModel is busy executing async tasks
+        this.queueMessageForDomain(domainName, parsedMessage.message);
+      }
     }
   }
 
+  private queueMessageForDomain(domainName: DomainName, message: JSONValue): void {
+    let queue = this.domainToMessageQueue.get(domainName);
+    if (!queue) {
+      queue = new Set();
+      this.domainToMessageQueue.set(domainName, queue);
+    }
+
+    queue.add(message);
+  }
+
+  private flushOutDomainMessagesQueues(): void {
+    for (const [domainName, queue] of this.domainToMessageQueue.entries()) {
+      for (const message of queue) {
+        this.dispatchMessageToDomainEventListeners(domainName, message);
+      }
+
+      queue.clear();
+    }
+  }
+
+  private isDomainMessagesQueueEmpty(domainName: DomainName): boolean {
+    const queue = this.domainToMessageQueue.get(domainName);
+    return queue === undefined || queue.size === 0;
+  }
```

Comment:
I think we can skip this check if the boolean flag is representative of the "readiness" (as mentioned above)